### PR TITLE
[SYCL] Check for aspect::fp64 in ESIMD/api/*ops_heavy.cpp

### PR DESCRIPTION
**Pull request by aelovikov-intel**

### CODE REVIEW

File: SYCL/ESIMD/api/bin_and_cmp_ops_heavy.cpp [hunk 1]
```diff
@@ -266,11 +266,13 @@ int main(void) {
...
   passed &= test<char, float, 7, BinOp, VEf, IDf>(arith_ops, q, 0.000001f);
-  passed &= test<short, double, 7, BinOp, VEf, IDf>(arith_ops, q, 1e-15);
+  if (dev.has(aspect::fp64))
+    passed &= test<short, double, 7, BinOp, VEf, IDf>(arith_ops, q, 1e-15);
   passed &= test<float, float, 32, BinOp, VEf, IDf>(arith_ops, q, 0.000001f);
   passed &= test<half, char, 1, BinOp, verify_n, IDf>(arith_ops, q, 1);
   passed &= test<half, unsigned int, 32, BinOp, VSf, IDf>(arith_ops, q, 1);
-  passed &= test<double, half, 7, BinOp, VSf, IDf>(arith_ops, q);
+  if (dev.has(aspect::fp64))
+    passed &= test<double, half, 7, BinOp, VSf, IDf>(arith_ops, q);
```
Comment (v-klochkov) [inline SYCL/ESIMD/api/bin_and_cmp_ops_heavy.cpp:275]: I think, this code should check for both fp64 and fp16

File: SYCL/ESIMD/api/bin_and_cmp_ops_heavy.cpp [hunk 2]
```diff
@@ -282,13 +284,15 @@ int main(void) {
   passed &= test<unsigned char, int, 1, BinOp, VSf, IDf>(div_op, q);
   passed &= test<char, float, 7, BinOp, VEf, IDf>(div_op, q, 0.000001f);
 #ifndef WA_BUG
-  passed &= test<short, double, 7, BinOp, VSf, IDf>(div_op, q);
+  if (dev.has(aspect::fp64))
+    passed &= test<short, double, 7, BinOp, VSf, IDf>(div_op, q);
 #endif // WA_BUG
   passed &= test<float, float, 32, BinOp, VEf, IDf>(div_op, q, 0.000001f);
   passed &= test<half, char, 1, BinOp, verify_n, IDf>(div_op, q, 1);
   passed &= test<half, unsigned int, 32, BinOp, VSf, IDf>(div_op, q, 1);
```
Comment (v-klochkov) [inline SYCL/ESIMD/api/bin_and_cmp_ops_heavy.cpp:292]: Would you please add check for fp16?

Comment (aelovikov-intel) [reply SYCL/ESIMD/api/bin_and_cmp_ops_heavy.cpp:292]: I'd rather do it in a different PR. Fp64 is way more urgent because our CI is broken.

**Comment by aelovikov-intel** [comment]
The tests modified in this PR start to pass in the CI (no mention of them in the failed tasks) hence the failures are unrelated.

FP16 aspect isn't as urgent and should be addressed in a different PR. Can anybody approve and merge this?

**Review by v-klochkov — APPROVED** [review]
Thank you for doing these changes.
Approved to speedup CI recovery. fp16 still needs similar checks later.

**Comment by aelovikov-intel** [comment]
@intel/llvm-gatekeepers we've agreed offline to proceed with this approach in the short term at least. The tests updated by this PR are passing in pre-commit CI (and thus other failures are unrelated).

Please merge this in.

**Pull request merged**